### PR TITLE
chore: fix warnings when compiling macOS client with either Xcode or CMake

### PR DESCRIPTION
**Pull request by nevack**
Every commit can be reviewed separately and contains explanation in message.
If one of commits is not something wanted, I can drop it.

### CODE REVIEW

**Comment by nevack** [comment]
@Coeur FYI [`5c4d8dd` (#6676)](https://github.com/transmission/transmission/pull/6676/commits/5c4d8dd5df5cc5dfa2701ac3b2464f38ce9f5831)

**Comment by Coeur** [comment]
I'm against using epsilon comparisons when it should be exact comparisons.
So I'll not approve yet what I was disagreeing with in #4400 review.
But give me some time/days and I'll try to re-wire my mind about those specific cases.

**Comment by tearfur** [comment]
Hmm, with 01ae195, I might be able to refactor #5702 to use a singleton class now. 🤔

**Review by ckerr — APPROVED** [review]
LGTM

**Comment by nevack** [comment]
> But give me some time/days and I'll try to re-wire my mind about those specific cases.

@Coeur Any news on that? 😃

**Comment by Coeur** [comment]
> > But give me some time/days and I'll try to re-wire my mind about those specific cases.
> 
> @Coeur Any news on that? 😃

This **weekend** I'll check. But we may have some more important pull requests to consider in priority @ckerr:
- a potential build fix: #6679
- the fix for the one and only known blocker for having a Transmission 4.0.6 release and a Transmission 4.1.0 BETA release: #5263
- fix miniupnpc: #6459
- fix a crash in stats: #5992
- fix data corruption on completion: #4178
- updating the generated web files: #6540

No mentioning the other features which are ready to merge for quite some time.

**Comment by ckerr** [comment]
@Coeur thank you for the list :green_heart: 

@nevack :+1:

File: macosx/InfoOptionsViewController.mm [hunk 1]
```diff
@@ -295,7 +296,7 @@ - (void)updateOptions
             checkRatio = kInvalidValue;
         }
 
-        if (!multipleRatioLimits && ratioLimit != torrent.ratioLimit)
+        if (!multipleRatioLimits && isRatioEqual(ratioLimit, torrent.ratioLimit))
```
Comment (Coeur) [inline macosx/InfoOptionsViewController.mm:299]: Should be `!isRatioEqual` instead of `isRatioEqual`.

@ckerr you merged before the weekend review. -_-

**Pull request merged**